### PR TITLE
docs

### PR DESCRIPTION
**Pull request by 0x00b1**
Basic Sphinx autodoc scaffolding.

I needed this to help determine what has and has not been implemented!

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    "sphinx.ext.autodoc",
+    "sphinx.ext.autosummary",
+    "sphinx.ext.doctest",
+    "sphinx.ext.duration",
+    "sphinx.ext.intersphinx",
+    "sphinx.ext.mathjax",
+    "sphinx.ext.napoleon",
+    "sphinx.ext.todo",
+    "sphinx.ext.viewcode",
+    "sphinx_copybutton"
```
Comment (mmuckley) [inline docs/source/conf.py:45]: With type annotations I've found `sphinx_autodoc_typehints` to be very nice. We have type annotations in the repository, so we might as well use this.

File: docs/source/conf.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import pytorch_sphinx_theme
+
+# -- Project information -----------------------------------------------------
+
+project = "neuralcompression"
+copyright = "2021, Facebook"
+author = "Facebook"
+
+# The full version, including alpha/beta/rc tags
+release = "0.1.0"
```
Comment (mmuckley) [inline docs/source/conf.py:27]: Could we import the version from `neuralcompression`?

**Review by mmuckley — CHANGES_REQUESTED** [review]
Overall looks good, just a couple of nits. I think we just need to set up some webhooks and this will be live on ReadtheDocs.

File: docs/source/conf.py (hunk 1 above)
Comment (0x00b1) [reply docs/source/conf.py]: Done!

File: docs/source/conf.py (hunk 2 above)
Comment (0x00b1) [reply docs/source/conf.py]: Done!

**Pull request merged**